File: Java/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        if (target < nums[0]) {
            return 0;
        }
        if (target > nums[nums.length - 1]) {
            return nums.length;
        }
        
        for (int i = 0; i < nums.length; i++) {
            if (target == nums[i]) {
                return i;
            } 
            if (i < nums.length - 1) {
                if (target > nums[i] && target < nums[i + 1]) {
                    return i + 1;
                }
            }
        }
        return -1;
    }
}

File: Kotlin/search-insert-position.kt
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        if (nums == null || nums.size == 0) return 0
        if (nums[0] >= target) return 0
        if (nums[nums.size - 1] < target) return nums.size
        
        var start = 0
        var end = nums.size - 1
        while (start < end) {
            var middle = (end + start) / 2
            if (nums[middle] == target) {
                return middle
            } else if (nums[middle] > target) {
                end = middle
            } else {
                start = middle + 1
            }
        }
        return start
    }
}
